src/explore-education-statistics-admin/src/components/editable/__tests__/EditableEmbedForm.test.tsx

```diff
@@ -0,0 +1,121 @@
+      />,
+    );
+
+    expect(screen.getByLabelText('Title')).toHaveValue('Dashboard title');
+    expect(screen.getByLabelText('URL')).toHaveValue(
+      'https://department-for-education.shinyapps.io/test-dashboard',
+    );
+    expect(screen.getByRole('button', { name: 'Save' })).toBeInTheDocument();
+    expect(screen.getByRole('button', { name: 'Cancel' })).toBeInTheDocument();
+  });
+
+  test('shows a validation error when no title is set', async () => {
+    render(<EditableEmbedForm onCancel={noop} onSubmit={noop} />);
+
+    userEvent.click(screen.getByLabelText('Title'));
+    userEvent.tab();
+
+    await waitFor(() => {
+      expect(screen.getByText('There is a problem')).toBeInTheDocument();
+      expect(
+        screen.getByRole('link', {
+          name: 'Enter a title',
+        }),
+      ).toBeInTheDocument();
+    });
+  });
+
+  test('shows a validation error when no url is set', async () => {
+    render(<EditableEmbedForm onCancel={noop} onSubmit={noop} />);
+
+    userEvent.click(screen.getByLabelText('URL'));
+    userEvent.tab();
+
+    await waitFor(() => {
+      expect(screen.getByText('There is a problem')).toBeInTheDocument();
+      expect(
+        screen.getByRole('link', {
+          name: 'Enter a URL',
+        }),
+      ).toBeInTheDocument();
+    });
+  });
+
+  test('shows a validation error when the url is invalid', async () => {
+    render(<EditableEmbedForm onCancel={noop} onSubmit={noop} />);
+
+    userEvent.type(screen.getByLabelText('URL'), 'Not a url');
+    userEvent.tab();
+
+    await waitFor(() => {
+      expect(screen.getByText('There is a problem')).toBeInTheDocument();
+      expect(
+        screen.getByRole('link', {
+          name: 'Enter a valid URL',
+        }),
+      ).toBeInTheDocument();
+    });
+  });
+
+  test('shows a validation error when the url is not from an allowed domain', async () => {
+    render(<EditableEmbedForm onCancel={noop} onSubmit={noop} />);
+
+    userEvent.type(screen.getByLabelText('URL'), 'http://test.com');
+    userEvent.tab();
+
+    await waitFor(() => {
+      expect(
+        screen.getByRole('link', {
+          name: 'URL must be on a permitted domain',
+        }),
+      ).toBeInTheDocument();
+    });
+  });
+
+  test('calls `onSubmit` with the form values when submitted successfully', async () => {
+    const handleSubmit = jest.fn();
+    render(<EditableEmbedForm onCancel={noop} onSubmit={handleSubmit} />);
+
+    userEvent.type(screen.getByLabelText('Title'), 'Dashboard title');
+    userEvent.type(
+      screen.getByLabelText('URL'),
+      'https://department-for-education.shinyapps.io/test-dashboard',
+    );
+    userEvent.click(screen.getByRole('button', { name: 'Save' }));
+
+    await waitFor(() => {
+      expect(handleSubmit).toHaveBeenCalledWith(
+        {
+          title: 'Dashboard title',
+          url: 'https://department-for-education.shinyapps.io/test-dashboard',
+        },
+        expect.anything(),
```

Comment:
Just wondering why we need this?

Reply:
It's because we're using `useFormSubmit` which means as well as the things we expect `toHaveBeenCalledWith` picks up various methods related to forms (e.g. validateForm, resetForm), so the `expect.anything()` just means we don't have to worry about them.